### ISSUE TITLE
[1.0 breaking change] - consolidate compression options

### ISSUE THREAD

**Issue by dralley** [rpm-software-management/createrepo_c#363]
createrepo_c currently has two different switches for compression types, "compress-type" and "general-compress-type".

The latter applies to everything, and the former only applies to files that are *not* "primary", "filelists", and "other".  There is no great reason for this, there should be one compression type option which is applied uniformly to all metadata produced by the
"createrepo_c" tool.  Yum would have choked on this because of the special handling around "comps", but I believe it should no longer be a concern?

Further suggestions:

More aggressive - remove Bzip2 support, contingent on https://github.com/rpm-software-management/createrepo_c/issues/338#issuecomment-1525805415

**Comment by dralley** [comment]
@kontura I know you didn't want to go quite this far but I thought there was going to be some functionality deprecated with warnings?

**Comment by kontura** [comment]
Unfortunately I didn't find the time to do any additional work on createrepo_c and we needed to get the other changes released at least a bit in advance in case there are some problems.
Therefore this was postponed.

**Comment by mattiaverga** [comment]
Chiming in the discussion, as docs are not really clear (at least, for me): am I right to understand that `--xz` option is just a shorthand for `----general-compress-type=xz`? Or is it a shorthand for `--compress-type`?

**Comment by kontura** [comment]
It is a shorthand for just `--compress-type=xz` (so it doesn't affect "primary", "filelists", and "other" xml metadata).

When we will be unifying the compression options we should also remove the `--xz` option.

**Comment by ppisar** [comment]
That's your use case. My use case is many small repositories. A good default is very subjective. Do you think zstd will be the best in 10 years? Is zstd supported on RHEL 7? A good default does not last in time. No compression has the advantage that it avoids a risk of breaking a compatibility. Or we can make the default build-time configurable, so it's not our = upstream problem anymore.

**Comment by dralley** [comment]
>No compression has the advantage that it avoids a risk of breaking a compatibility.

"none" is not currently even an option in createrepo_c.  The maximally compatible choice would be to use gzip (the default prior to 1.0) forevermore, which I would still prefer to defaulting to no compression at all.

I do agree that a "none" choice should exist, but it should not be the default.  It has no compatibility advantages whatsoever over other options.

>Do you think zstd will be the best in 10 years?  Is zstd supported on RHEL 7? A good default does not last in time.

Does it really matter if it's "the best" in 10 years?  It's meaningfully better across relevant metrics and widely supported by both Red Hat and SUSE derived distributions for several years now.  Support was added to `libsolv` in 2018 and is inherited by both `dnf` and `zypper`.

But no, not RHEL 7 and "yum" generally.

**Comment by dralley** [comment]
Submitted https://github.com/rpm-software-management/createrepo_c/pull/411/ for `--compatibility` defaulting to gzip
